### PR TITLE
Adjust keepAlive for test under JDK 9 [ci: last-only]

### LINKED ISSUE #864: `scala-concurrent-tck` runs long
2.13.x, on my M1:

```
ok 1 - jvm/scala-concurrent-tck.scala          [duration 68.95s]
```

Credits to @Duhemm and Gradle's Develocity :)

Some recent PRs around that test
  - https://github.com/scala/scala/pull/8365
  - https://github.com/scala/scala/pull/8402
  - https://github.com/scala/scala/pull/9680

where I find "duration 6.85s" and "it takes about 13 seconds", so maybe something has changed in the meantime.

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes https://github.com/scala/scala-dev/issues/864

The test waits for the pool thread to terminate instead of interrupting it, probably to ensure that the thread wasn't planning on issuing spurious reports of exceptional circumstances.

The existing test used the default FJP available under JDK 8, which has a default keep-alive of 60 seconds, or one year in DX time.

This commit ~reflectively~ creates a FJP under JDK 9+ with a short keep-alive, or uses a simple thread pool executor otherwise.

### CODE REVIEW

**Comment by som-snytt** [comment]
@lrytz this is quite tiresome. My review is that it should skip the test on JDK 8, but the twist is that the offending sub-test should be split into a separate test file with `java: jdk9+`.

File: test/files/jvm/scala-concurrent-tck.scala [hunk 1]
```diff
@@ -1076,6 +1045,104 @@ class ExecutionContextPrepare extends TestBase {
...
+      def invokeAny[T](x$1: java.util.Collection[_ <: java.util.concurrent.Callable[T]], x$2: Long, x$3: java.util.concurrent.TimeUnit): T = ???
+      def invokeAny[T](x$1: java.util.Collection[_ <: java.util.concurrent.Callable[T]]): T = ???
+      def isShutdown(): Boolean = fjp.isShutdown
+      def isTerminated(): Boolean = fjp.isTerminated
+      def shutdown(): Unit = fjp.shutdown()
+      def shutdownNow(): java.util.List[Runnable] = fjp.shutdownNow()
+      def submit(r: Runnable): java.util.concurrent.Future[_] = fjp.submit(r)
+      def submit[T](task: Runnable, res: T): java.util.concurrent.Future[T] = fjp.submit(task, res)
+      def submit[T](task: java.util.concurrent.Callable[T]): java.util.concurrent.Future[T] = fjp.submit(task)
+    }
```
Comment (lrytz) [inline test/files/jvm/scala-concurrent-tck.scala]: Does `ExecutionContext.fromExecutorService(fjp, report(null, _))` not work as expected here?

**Review by lrytz — COMMENTED** [review]
> this is quite tiresome

I bet..!

LGTM in general.

File: test/files/jvm/scala-concurrent-tck.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/jvm/scala-concurrent-tck.scala:1110]: I'll make time for the extra-credit question.

Comment (som-snytt) [reply test/files/jvm/scala-concurrent-tck.scala:1110]: If I understand the question, the diff is that we can't specify the keep-alive for JDK 8, so the current behavior is the long delay waiting for the thread to exit. I think the way to go is my previous comment, just skip it on JDK 8.

**Comment by som-snytt** [comment]
the previous failure was spurious, so rebased.

**Comment by som-snytt** [comment]
I wonder if the spurious errors are due to using a spuriously ancient JDK.

```
[info] [info] compiling 1 Scala source and 1 Java source to /tmp/sbt_3b38204f/target/classes ...
[info] #
[info] # A fatal error has been detected by the Java Runtime Environment:
[info] #
[info] #  SIGBUS (0x7) at pc=0x00007f62579ddfad, pid=21674, tid=0x00007f61f7afc700
[info] #
[info] # JRE version: OpenJDK Runtime Environment (8.0_332-b09) (build 1.8.0_332-8u332-ga-1~deb9u1-b09)
[info] # Java VM: OpenJDK 64-Bit Server VM (25.332-b09 mixed mode linux-amd64 compressed oops)
[info] # Problematic frame:
[info] # C  [libzip.so+0x4fad]
[info] #
[info] # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
[info] #
[info] # An error report file with more information is saved as:
[info] # /tmp/sbt_3b38204f/hs_err_pid21674.log
[info] Compiled method (nm)   72113   85     n 0       java.util.zip.ZipFile::getEntry (native)
[info]  total in heap  [0x00007f6248b12b90,0x00007f6248b12f00] = 880
[info]  relocation     [0x00007f6248b12cb8,0x00007f6248b12d00] = 72
[info]  main code      [0x00007f6248b12d00,0x00007f6248b12f00] = 512
[info] #
[info] # If you would like to submit a bug report, please visit:
[info] #   http://bugreport.java.com/bugreport/crash.jsp
[info] # The crash happened outside the Java Virtual Machine in native code.
[info] # See problematic frame for where to report the bug.
[info] #
[error] x apiinfo/circular-structure 
[error]  Cause of test exception: {line 4}  Command failed
```

**Comment by som-snytt** [comment]
This might also be intermittent
```
# starting 45 tests in presentation
...............X.............................
!!  1 - presentation/doc                          [output differs][duration 8.90s]
% diff /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/presentation/doc.check /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/presentation/doc-presentation.log
@@ -1 +1,3 @@
 reload: Base.scala, Class.scala, Derived.scala
+Unexpected Derived class comment:None
+Unexpected foo method comment:None
```

**Comment by lrytz** [comment]
Yes, `presentation/doc` failing with `None` happens from time to time; probably worth a scala-dev ticket.

**Comment by som-snytt** [comment]
it's already known, infamously, as "the less famous flaky test".

**Comment by som-snytt** [comment]
I needed this today. I ran the test suite, just to make sure my computer fans still work (and check the check files), and the fans went silent but the suite was hung. Of course it was this problem, just waiting to time out.

**Comment by som-snytt** [comment]
> LGTM in general.

How about in particular?

I've considered that a PR should come with a "motivation", "how I fixed this", etc.

**Comment by lrytz** [comment]
> > 
> 
> How about in particular?
> 
> I've considered that a PR should come with a "motivation", "how I fixed this", etc.

You were clearly motivated and fixed it very well :)

**Review by lrytz — APPROVED** [review]

**Pull request merged**